src/proc/namer.rs

```diff
@@ -97,12 +91,9 @@ impl Namer {
         for (fun_handle, fun) in module.functions.iter() {
             let fun_name = this.call_or(&fun.name, "function");
             output.insert(NameKey::Function(fun_handle), fun_name);
-            if fun.parameter_types.len() >= MAX_PARAM_COUNT as usize {
-                unreachable!()
-            }
-            for (index, _) in fun.parameter_types.iter().enumerate() {
-                let name = format!("{}{}", PARAM_NAME, index + 1);
-                output.insert(NameKey::FunctionParameter(fun_handle, index as u32), name);
+            for (index, arg) in fun.arguments.iter().enumerate() {
+                let name = this.call_or(&arg.name, "param");
```

Comment:
Maybe suffix the fallback value in the following format: "param_[index]"? No real benefit, other than having unique values for the name.